internal/controller/services/auth/resources/data-science-metrics-admin-clusterrole.tmpl.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: data-science-metrics-admin
+  labels:
+    app.kubernetes.io/managed-by: opendatahub-operator
+    platform.opendatahub.io/part-of: auth
+rules:
```

Comment:
These permissions are far too broad. Please add justifications as comments to the yaml (what the permission used for), and remove anything not strictly needed.

Reply:
[The pattern](https://github.com/openshift/enhancements/blob/10e8fd8af459c4ad4757af386fec9f9f6145c72a/enhancements/monitoring/user-workload-monitoring.md#query-endpoint) UWM is using: [gated](https://github.com/openshift/cluster-monitoring-operator/blob/85ce8635536ab132fcf618de81a379ad7fff8cca/assets/thanos-querier/kube-rbac-proxy-secret.yaml#L16-L24) through access to [pods.metrics.k8s.io](http://pods.metrics.k8s.io/) in the requested namespace; this access is granted by default to users with edit clusterrole (and to admins); we shouldn't have to manage roles/rolebindings for this.